### PR TITLE
Get acloudw port number and register a cloud based VM to DUT cache list

### PR DESCRIPTION
**Pull request by 4n12i**
- Get a port number from output of `acloudw` command 
- Register a VM to DUT cache list
- Suggest how to connect to a virtual DUT

### CODE REVIEW

File: src/cmd/vm.rs [hunk 1]
```diff
@@ -379,18 +387,38 @@ fn run_acloudw_cmd(opts: &[&str]) -> Result<()> {
...
 
-    Ok(())
+    Ok(port)
+}
+
+fn get_acloudw_port_number(r: impl BufRead) -> Result<String> {
+    let mut port = String::new();
+    let split_iter = r
+        .split(b'\n')
+        .map(|l| String::from_utf8_lossy(&strip_ansi_escapes::strip(l.unwrap())).to_string());
```
Comment (4n12i) [inline src/cmd/vm.rs:412]: Use `strip_ansi_escapes` crate to remove ANSI escape sequences (color code) from output. 

https://docs.rs/strip-ansi-escapes/0.2.0/strip_ansi_escapes/

Comment (hikalium) [reply src/cmd/vm.rs:416]: I think it is a good idea to add a comment in the code to explain why removing the ANSI escape sequence here is needed (like `Remove the escape sequences here to make the regexp search work` or something)

File: src/cmd/vm.rs [hunk 2]
```diff
@@ -236,23 +244,24 @@ pub struct ArgsStart {
...
+        "9222".to_string()
+    };
 
-        println!(
-            "You can connect the VM instance with `lium dut shell --dut {:?}`.",
-            info.id()
-        );
-        println!("To push an Android build a betty VM, run `lium arc flash`.");
-    }
+    let info = register_dut(&format!("127.0.0.1:{port}"))?;
```
Comment (hikalium) [inline src/cmd/vm.rs:254]: Maybe adding a comment why 127.0.0.1 is here is helpful for the users!

File: src/cmd/vm.rs [hunk 3]
```diff
@@ -364,7 +372,7 @@ fn get_cheeps_image_name(config: &Config, is_container: bool, branch: &str) -> R
     Ok(cheeps)
 }
 
-fn run_acloudw_cmd(opts: &[&str]) -> Result<()> {
+fn run_acloudw_cmd(opts: &[&str]) -> Result<String> {
```
Comment (hikalium) [inline src/cmd/vm.rs:375]: Port number is a 16-bit unsigned number, so the return value type should be u16, not a String.

File: src/cmd/vm.rs [hunk 4]
```diff
@@ -379,18 +387,42 @@ fn run_acloudw_cmd(opts: &[&str]) -> Result<()> {
...
 
     if !result.success() {
         error!("acloudw failed")
     }
 
-    Ok(())
+    Ok(port)
+}
+
+fn get_acloudw_port_number(r: impl BufRead) -> Result<String> {
```
Comment (hikalium) [inline src/cmd/vm.rs:412]: Also please add a function-level documentation about the return value.

File: src/cmd/vm.rs [hunk 5]
```diff
@@ -236,23 +244,24 @@ pub struct ArgsStart {
...
+        "You can connect the VM instance with `lium dut shell --dut {:?}`.",
+        info.id()
+    );
+    println!("To push an Android build a betty VM, run `lium arc flash`.");
 
     Ok(())
 }
 
-fn run_acloudw(args: &ArgsStart) -> Result<()> {
+fn run_acloudw(args: &ArgsStart) -> Result<String> {
```
Comment (hikalium) [inline src/cmd/vm.rs:264]: Add a comment to explain what is the return value (and change it to u16)

**Review by hikalium — CHANGES_REQUESTED** [review]

File: src/cmd/vm.rs (hunk 1 above)
Comment (4n12i) [reply src/cmd/vm.rs:422]: Thank you for the advice! 
I added a comment.

File: src/cmd/vm.rs (hunk 2 above)
Comment (4n12i) [reply src/cmd/vm.rs:256]: Added a comment, thank you!

File: src/cmd/vm.rs (hunk 3 above)
Comment (4n12i) [reply src/cmd/vm.rs]: Indeed, I updated it.

File: src/cmd/vm.rs (hunk 4 above)
Comment (4n12i) [reply src/cmd/vm.rs]: Added a comment!

File: src/cmd/vm.rs (hunk 5 above)
Comment (4n12i) [reply src/cmd/vm.rs]: Updated the return value and added a comment!

**Review by hikalium — APPROVED** [review]

**Comment by hikalium** [comment]
Sorry for the late reply... I think the change looks good to me for now! Thank you!

**Pull request merged**